bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/renault/api/MyRenaultHttpSession.java

```diff
@@ -0,0 +1,266 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.renault.internal.renault.api;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.util.Fields;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.binding.renault.internal.RenaultConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Translation of this python code: https://github.com/hacf-fr/renault-api
+ *
+ * @author Doug Culnane - Initial contribution
+ */
+public class MyRenaultHttpSession {
+
+    private HttpClient httpClient;
+    private Constants constants;
+
+    private Car car;
+    private String kamereonToken;
+    private String kamereonaccountId;
+    private String cookieValue;
+    private String personId;
+    private String gigyaDataCenter;
+    private String jwt;
+
+    private final Logger logger = LoggerFactory.getLogger(MyRenaultHttpSession.class);
+
+    public MyRenaultHttpSession(RenaultConfiguration config) throws Exception {
+
+        car = new Car();
+        this.httpClient = new HttpClient(new SslContextFactory(true));
+        this.constants = new Constants(config.locale);
+
+        httpClient.start();
+        login(config);
+        getAccountInfo(config);
+        getJWT(config);
+        getAccountID(config);
+        getVehicle(config);
```

Comment:
Maybe it would be better to move these calls in a separate method and keep a constructor that just init variables and throws no exception.